lib/wasix/src/runtime/task_manager/tokio.rs

```diff
@@ -253,17 +253,27 @@ impl VirtualTaskManager for TokioTaskManager {
         } else {
             tracing::trace!("spawning task_wasm in blocking thread");
 
+            let (sx, rx) = std::sync::mpsc::channel();
+
             // Run the callback on a dedicated thread
             self.pool.execute(move || {
                 tracing::trace!("task_wasm started in blocking thread");
-                let (mut ctx, mut store) = WasiFunctionEnv::new_with_store(
+                let (mut ctx, mut store) = match WasiFunctionEnv::new_with_store(
```

Comment:
nit: I find code like this super unreadable - I always do 

```
let res = ...;
match res { ...
```

Reply:
[dc81648](https://github.com/wasmerio/wasmer/pull/5440/commits/dc816488050c232e3ef9179302d067d818b1afb2)